Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -117,22 +116,265 @@ void cppMain() {
 		displacement_inches = lin_pot->DisplacementInches();
 
 
-		printf("\n Percentage: %f", displacement_inches);
+		//printf("\n Percentage: %f", displacement_inches);
+
 		if(CANMessageReceivedFlag){
 			__disable_irq();
-			TxData[0] = 1 ;
-			TxData[1] = 2 ;
-			TxData[2] = 3 ;
-			TxData[3] = RxData[3] ;
-			TxData[4] = RxData[4] ;
-			TxData[5] = 6 ;
-			TxData[6] = 0x12 ;
-			TxData[7] = (RxData[4] * 256 + RxData[3]) * .1;
-
-			HAL_CAN_AddTxMessage(&hcan1, &TxHeader,TxData,&TxMailbox);
-			CANMessageReceivedFlag = false;
+			canParseDataFrame();
+			if(CanID == 0x0CFFF048){
+				printf("\t\t 0x0CFFF048 \n");
+				printf("\r");
+				printf("RPM = %.2f ", DecodeDataFrame[0]);
+				printf("TPS = %.2f ", DecodeDataFrame[2]);
+				printf("Fuel Open Time = %.2f ", DecodeDataFrame[4]);
+				printf("Ignition Angle = %.2f \n", DecodeDataFrame[6]);
+				printf("\r");
+			}
+			if(CanID == 0x0CFFF548){
+				printf("\t\t 0x0CFFF548 \n");
+				printf("\r");
+				printf("Battery Volt = %.2f ", DecodeDataFrame[0]);
+				printf("Air Temp = %.2f ", DecodeDataFrame[2]);
```

Comment:
![can snippet ](https://github.com/DallasFormulaRacing/DataAcquisition2.0/assets/113472670/e4337fa6-85c2-41f2-adad-89e4ce18f091)

Here is what the serial monitor looked like. The data shown on the serial monitor seems to support my testing but obviously in person testing with the ECU is needed to fully validate if the CAN parsing is correct